tests/conftest.py

```diff
@@ -8,3 +9,11 @@
 def test_app():
     client = TestClient(app)
     yield client
+
+
+@pytest.fixture()
+def mock_failed_connection_httpx_get():
+    async def _mock_httpx_get_with_connect_error(self, **kwargs):
```

Comment:
:eyes:
a) like **kwargs here, I think we could use that with other fixtures where we're less concerned with the args
b) why `self` here?

Reply:
See new commits for comments added to explain this!